WordPress/gutenberg
# Improve accessibility of the Warning component in the block editor

Accessibility Fix: #62737

## What?
This PR improves the Warning component in the block editor by enhancing its accessibility.

## Why?
Currently, the Warning component only provides a visual cue without notifying screen readers, which limits accessibility. These changes ensure that the component is accessible to all users.

## How?

- Added ARIA role="alert" and aria-live="assertive" to the Warning component for screen reader announcements.
- Added aria-describedby="block-editor-warning-message" to the Warning actions.
- Added tabIndex="0", so that the Warning can be accessed using tabs. This helps to navigate inside the warning.

## Testing Instructions

1. Add a More block to a post and copy it using the keyboard shortcut or the Options menu.
2. Paste the copied block elsewhere in the post.
3. Observe the Warning message appears with improved design and triggers a screen reader announcement.
